lib/puppet/type/mount.rb

```diff
@@ -293,5 +297,18 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(self[:name]) }
+
+    # Autobefore the mount point's child file paths
+    autobefore(:file) do
+      dependencies = []
+      file_resources = catalog.resources.select { |resource| resource.type == :file }
+      children_file_resources = file_resources.select { |resource| resource[:path].to_s =~ %r(#{self[:name]}/.) }
```

Comment:
This may not match if the file `:path` isn't canonicalized, e.g. `/foo/bar/../baz`. You may be able to do `File.expand_path(resource[:path]) =~ %r(#{self[:name]}/.)`